internal/clusterinfo/data.go

```diff
@@ -548,7 +549,16 @@ func (c *ClusterInfo) GetNSQDStats(producers Producers, selectedTopic string) ([
 			defer wg.Done()
 
 			addr := p.HTTPAddress()
-			endpoint := fmt.Sprintf("http://%s/stats?format=json", addr)
+
+			var endpoint string
+			if len(selectedChannel) != 0 {
```

Comment:
This isn't entirely correct, as it doesn't test for `selectedTopic`, let's just adopt the code in #925 wholesale for this block:

```go
if selectedTopic == "" {
	endpoint = fmt.Sprintf("http://%s/stats?format=json", addr)
} else if selectChannel == "" {
	endpoint = fmt.Sprintf("http://%s/stats?topic=%s&format=json", addr, selectedTopic)
} else {
	endpoint = fmt.Sprintf("http://%s/stats?topic=%s&channel=%s&format=json", addr, selectedTopic, selectChannel)
}
```

Reply:
Let me humbly offer my own variant here:

```go
endpoint := fmt.Sprintf("http://%s/stats?format=json", addr)
if selectedTopic != "" {
        endpoint += "&topic=" + selectedTopic
        if selectedChannel != "" {
                endpoint += "&channel=" + selectedChannel
        }
}
```

Reply:
👍

Reply:
Thanks for your review and the more simplified and clearer refactor.